lib/chef/provider/cron.rb

```diff
@@ -51,6 +67,9 @@ def load_current_resource
         @current_resource = Chef::Resource::Cron.new(@new_resource.name)
         @current_resource.user(@new_resource.user)
         @cron_exists = false
+        # Raise an execption if Chef can't read or update the crontab.
+        can_modify_cron?
```

Comment:
this should be moved into define_resource_requirements and should support why-run mode.   so when the user doesn't exist it should result in a "assuming user would have been created" style of warning in why-run.  the other exception should most likely also fail in why-run mode, but should still be raised in d_r_r.